great_expectations/data_context/data_context.py

```diff
@@ -2325,26 +2375,27 @@ def _list_ymls_in_checkpoints_directory(self):
         files = glob.glob(os.path.join(checkpoints_dir, "*.yml"), recursive=False)
         return files
 
-    def _save_project_config(self):
-        """Save the current project to disk."""
-        logger.debug("Starting DataContext._save_project_config")
+    def add_store(
+        self, store_name: str, store_config: dict
+    ) -> Union[StoreBackend, Store, None]:
+        logger.debug("Starting DataContext.add_store for store %s" % store_name)
 
-        config_filepath = os.path.join(self.root_directory, self.GE_YML)
-        with open(config_filepath, "w") as outfile:
-            self._project_config.to_yaml(outfile)
+        new_store: Union[StoreBackend, Store, None] = super().add_store(
+            store_name, store_config
+        )
 
-    def add_store(self, store_name, store_config):
-        logger.debug("Starting DataContext.add_store for store %s" % store_name)
+        if store_name is not None:
+            self._save_project_config()
 
-        new_store = super().add_store(store_name, store_config)
-        self._save_project_config()
         return new_store
 
-    def add_datasource(self, name, **kwargs):
+    def add_datasource(self, name: str, **kwargs) -> Datasource:
         logger.debug("Starting DataContext.add_datasource for datasource %s" % name)
 
         new_datasource = super().add_datasource(name, **kwargs)
-        self._save_project_config()
+
+        if name is not None:
```

Comment:
Should there be a warning or exception here if the `name is None`?

Reply:
In my mind -- no, because "None" is a legitimate value; we just do not want to attempt to save it.  Happy to discuss.

Reply:
Makes sense! So if `None` is allowed for parameter `name` should the function signature add Optional (https://docs.python.org/3.7/library/typing.html#typing.Optional):
```suggestion
    def add_datasource(self, name: Optional[str], **kwargs) -> Datasource:
```